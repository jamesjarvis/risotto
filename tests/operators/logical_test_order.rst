fun bool false_fun() {
    print "false_fun";
    return false;
}

fun bool true_fun() {
    print "true_fun";
    return true;
}

if(false_fun() || true_fun()) {
    print "or > test1";
}

if(true_fun() || false_fun()) {
    print "or > test2";
}

if(true_fun() || true_fun()) {
    print "or > test3";
}

if(false_fun() || false_fun()) {
    print "or > test4";
}



if(false_fun() && true_fun()) {
    print "and > test1";
}

if(true_fun() && false_fun()) {
    print "and > test2";
}

if(true_fun() && true_fun()) {
    print "and > test3";
}

if(false_fun() && false_fun()) {
    print "and > test4";
}
