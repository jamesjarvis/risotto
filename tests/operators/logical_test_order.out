false_fun
true_fun
or > test1
true_fun
or > test2
true_fun
or > test3
false_fun
false_fun
false_fun
true_fun
false_fun
true_fun
true_fun
and > test3
false_fun
